### PR TITLE
Feat/add properties to button component

### PR DESCRIPTION
**Pull request by ErikssonJoakim**
This PR implements #379

### CODE REVIEW

File: src/ui/atoms/button/button.scss [hunk 1]
```diff
@@ -12,19 +12,15 @@ $background-colors: (
...
-    justify-content: space-around;
-    align-items: center;
+
     min-width: 65px;
     max-width: 273px;
     overflow: hidden;
 
     &.variant-secondary {
-      border-radius: 50px;
+      border-radius: 50%;
```
Comment (fredericvilcot) [inline src/ui/atoms/button/button.scss:23]: By doing this, the `secondary` variant is always rounded... 
image

```suggestion
      border-radius: 24px;
```

File: src/ui/atoms/button/Button.tsx [hunk 2]
```diff
@@ -42,22 +54,34 @@ export const Button: React.FC<TButtonProps> = ({
...
   onClick
 }: TButtonProps): JSX.Element => (
   <button
     className={classNames('okp4-button-main', {
       [`variant-${variant}`]: true,
       [`size-${size}`]: true,
       [`background-color-${backgroundColor}`]: true,
-      disabled: !!disabled
+      disabled,
+      icon: variant === 'icon'
```
Comment (fredericvilcot) [inline src/ui/atoms/button/Button.tsx:68]: Why add a `icon` className as the `variant-icon` className already exists? 🧐

File: src/ui/atoms/button/Button.tsx [hunk 3]
```diff
@@ -42,22 +54,34 @@ export const Button: React.FC<TButtonProps> = ({
...
     title={label}
     type="button"
   >
-    <Typography as="div" color="highlighted-text" fontSize="small" fontWeight="light">
-      {label}
-    </Typography>
+    {variant === 'icon' ? (
+      icon
+    ) : (
+      <div className="okp4-button-label">
```
Comment (fredericvilcot) [inline src/ui/atoms/button/Button.tsx:78]: The className `okp4-button-label` should carry more context in its naming as it does not only refers to the label

File: stories/atoms/components/button/Button.stories.mdx [hunk 4]
```diff
@@ -174,3 +186,38 @@ The `Button` component can be disabled at any time, thus preventing any user int
...
+        style={{
+          display: 'flex',
+          alignItems: 'center',
+          justifyContent: 'center',
+          gap: '25px',
+          flexWrap: 'wrap'
+        }}
+      >
+        <Button icon={<Icon name="wallet" />} label="wallet" variant="icon" />
+        <Button leftIcon={<Icon name="wallet" />} label="Get wallet address" />
```
Comment (fredericvilcot) [inline stories/atoms/components/button/Button.stories.mdx:214]: Could you decrease a bit the icon sizes please? (30px looks too big in the button)

File: src/ui/atoms/button/button.scss [hunk 5]
```diff
@@ -40,14 +36,42 @@ $background-colors: (
 
     &.size-small {
       padding: 4px 10px;
+
+      &.icon {
+        padding: 7px;
```
Comment (fredericvilcot) [inline src/ui/atoms/button/button.scss:41]: There is some inconsistencies between small/medium and large paddings, I will have uniformed paddings more consistently: 8px for small, 12px for medium and 16px for large (keep a 4px step)

File: src/ui/atoms/button/button.scss [hunk 6]
```diff
@@ -40,14 +36,42 @@ $background-colors: (
...
+      max-width: unset;
+      font-size: 0;
+      border-radius: 50%;
+    }
+
+    .okp4-button-label {
+      display: grid;
+      grid-auto-flow: column;
+      gap: 10px;
+      justify-content: space-around;
```
Comment (fredericvilcot) [inline src/ui/atoms/button/button.scss:73]: I'm not sure this rule has some effect here

File: src/ui/atoms/button/button.scss [hunk 7]
```diff
@@ -40,14 +36,42 @@ $background-colors: (
...
+      min-width: unset;
+      max-width: unset;
+      font-size: 0;
+      border-radius: 50%;
+    }
+
+    .okp4-button-label {
+      display: grid;
+      grid-auto-flow: column;
+      gap: 10px;
```
Comment (fredericvilcot) [inline src/ui/atoms/button/button.scss:72]: I will have added a little more gap between elements to better fit the wireframes (the goal is also not to have as much space as in the wireframe):
image

**Review by fredericvilcot — CHANGES_REQUESTED** [review]
Good, some little changes to increase consistency and UX 😌

File: src/ui/atoms/button/button.scss (hunk 1 above)
Comment (ErikssonJoakim) [reply src/ui/atoms/button/button.scss:23]: Unfortunate miss by me, will fix 👍

File: src/ui/atoms/button/Button.tsx (hunk 2 above)
Comment (ErikssonJoakim) [reply src/ui/atoms/button/Button.tsx:68]: Good question 😪  Another miss

File: stories/atoms/components/button/Button.stories.mdx (hunk 4 above)
Comment (ErikssonJoakim) [reply stories/atoms/components/button/Button.stories.mdx:214]: How about if I decrease it to 25px? I find that if I decrease it too much the details of the icon become too small.

File: src/ui/atoms/button/button.scss (hunk 7 above)
Comment (ErikssonJoakim) [reply src/ui/atoms/button/button.scss:72]: I could add some more gap and if the user wants to add more spacing they can just increase the width and position it where they want :)

File: src/ui/atoms/button/Button.tsx (hunk 3 above)
Comment (ErikssonJoakim) [reply src/ui/atoms/button/Button.tsx]: You are write, I've changed it too `okp4-button-description`. It's hard to come up with a more precise name as the content rendering depends on if the property values exists or not.

**Review by fredericvilcot — APPROVED** [review]
Sounds good, great 👍

**Review by JoeyDjr — APPROVED** [review]

**Review by lolottetheclash — APPROVED** [review]
Nice work😉

**Review by ccamel — APPROVED** [review]
👍

**Pull request merged**